nautobot/utilities/views.py

```diff
@@ -151,3 +187,1051 @@ def get_return_url(self, request, obj=None):
+                "fields": self.import_form().fields,
+                "obj_type": self.import_form._meta.model._meta.verbose_name,
+                "return_url": self.get_return_url(request),
+                "active_tab": "csv-data",
+            },
+        )
+
+    def handle_bulk_import_post(self, request):
+        logger = logging.getLogger("nautobot.views.BulkImportView")
+        new_objs = []
+        form = self._import_form_for_bulk_import(request.POST)
+
+        if form.is_valid():
+            logger.debug("Form validation was successful")
+
+            try:
+                # Iterate through CSV data and bind each row to a new model form instance.
+                with transaction.atomic():
+                    headers, records = form.cleaned_data["csv_data"]
+                    for row, data in enumerate(records, start=1):
+                        obj_form = self.import_form(data, headers=headers)
+                        restrict_form_fields(obj_form, request.user)
+
+                        if obj_form.is_valid():
+                            obj = self._save_obj_for_bulk_import(obj_form, request)
+                            new_objs.append(obj)
+                        else:
+                            for field, err in obj_form.errors.items():
+                                form.add_error("csv", f"Row {row} {field}: {err[0]}")
+                            raise ValidationError("")
+
+                    # Enforce object-level permissions
+                    if self.queryset.filter(pk__in=[obj.pk for obj in new_objs]).count() != len(new_objs):
+                        raise ObjectDoesNotExist
+
+                # Compile a table containing the imported objects
+                obj_table = self.table(new_objs)
+
+                if new_objs:
+                    msg = f"Imported {len(new_objs)} {new_objs[0]._meta.verbose_name_plural}"
+                    logger.info(msg)
+                    messages.success(request, msg)
+
+                    return render(
+                        request,
+                        "import_success.html",
+                        {
+                            "table": obj_table,
+                            "return_url": self.get_return_url(request),
+                        },
+                    )
+
+            except ValidationError:
+                pass
+
+            except ObjectDoesNotExist:
+                msg = "Object import failed due to object-level permissions violation"
+                logger.debug(msg)
+                form.add_error(None, msg)
+
+        else:
+            logger.debug("Form validation failed")
+
+        return render(
+            request,
+            self.template_name,
+            {
+                "form": form,
+                "fields": self.import_form().fields,
+                "obj_type": self.import_form._meta.model._meta.verbose_name,
+                "return_url": self.get_return_url(request),
+            },
+        )
```

Comment:
This is not something I would like to see us keep around, because it's not leveraging the built-in [`TemplateHTMLRenderer`](https://www.django-rest-framework.org/api-guide/renderers/#templatehtmlrenderer). And is instead using manual rendering.

Ideally we should be able to arrive at a solution where we can use the SAME viewset classes for both API and UI and have the presentation be a matter for the renderer class used at call time.